### DISCUSSION TITLE
Adapting the Indentation-Sensitive Languages example for implicit whitespace

### DISCUSSION THREAD

**Discussion by ElnuDev** [pest-parser/pest#861] [Q&A] [answered]
I'm trying to adapt the [Indentation-Sensitive Languages example in Chapter 3.2 of the Book](https://pest.rs/book/grammars/syntax.html#indentation-sensitive-languages) and it breaks when implicit whitespace is defined. If I define `WHITESPACE = _{ " " }` then all of the nested blocks are absent in the resulting AST, as presumably this would "eat" all of the indentation. I've messed around a bit with the example but I haven't found a solution. Any ideas? I'm pretty new to pest. Thanks in advance!

**Comment by tomtau** [comment, marked as answer]
You can potentially prevent implicit whitespaces by using atomic rules: https://pest.rs/book/grammars/syntax.html#atomic

FYI @reivilibre wrote that section about indentation-sensitive languages (and may chime in).

**Reply by jbmwdzg** [reply]
@tomtau  where can find such section about indentation-sensitive languages? thanks.
I am working on the development of an indentation-sensitive languages and really want to have a reference.

**Reply by tomtau** [reply]
@jbmwdzg https://pest.rs/book/grammars/syntax.html#indentation-sensitive-languages

**Reply by jbmwdzg** [reply]
Thanks @tomtau
